### DISCUSSION TITLE
Unauthorized when writing to Influxdb V2.6.1

### DISCUSSION THREAD

**Discussion by ZonPV** [johanmeijer/grott#270] [Q&A] [answered]
Trying to upgrade all devices to Influx V2. It runs in a separate container on my Synology on a different port (8087). Works for Domoticz. So now Grottserver. Dowloaded version 2.7.8.
Created a bucket named Grott and API token with full access (R/W) for that bucket.
This is the part of the grott.ini
```
[influx]
# Influxdb parameters definitions V2
influx = true
influx2 = true
ip = 192.168.x.xx
port = 8087
token  = "0UtNi7DBVXIY8LExxxxxxxxxxxxxx"
org  = "xxxxxxxxx"
bucket = "Grott" 
```
Beware the portnumber, it's ok because I read errors in the log of that container!

This is the error in the log of **Grottserver:**
```
 - Grott InfluxDB V2 initiating started
 - Grott error: can not contact InfluxDB
(401)
Reason: Unauthorized

HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json; charset=utf-8', 'X-Influxdb-Build': 'OSS', 'X-Influxdb-Version': 'v2.6.1', 'X-Platform-Error-Code': 'unauthorized', 'Date': 'Thu, 26 Jan 2023 10:10:49 GMT', 'Content-Length': '55'})

HTTP response body: {"code":"unauthorized","message":"unauthorized access"}
Grott Influxdb initialisation error
```

The only error in the Influx container is:
`2023-01-26T10:41:36.904161Z	info	Unauthorized	{"log_id": "0fb_V_zl000", "error": "authorization not found"}
`
Wondering about configuration in Grott.ini:
Is the config correct? 2 times Influx = true?

Not familiar with Influx so any help appreciated.
Thanks in advance!

**Comment by johanmeijer** [comment]
Yes 2x times influx is oké first you specify that you want to send to influx and the second one (influx2=true) that you want to use inluxdb Version 2.x (and not Version 1).

I still use influx Version 1 in my daily environment. I have tot set up the influx 2 again to see what is going wrong. 

8087 is the right port? 8086 seems to be the default for influx.

**Reply by ZonPV** [reply]
@johanmeijer : thanks for reply.
Thank you for confirmation about 2 times naming influx.
Sure about port 8087 as I run Influx 1.8 in different container with 8086 during migration for other (old) devices as my Omnik. 

Domoticz uses now V2 with port 8087 for datapush to Influx and it works perfect.

Double checked my username and Organiszation Profile name: it's both all in lower case. 

I give it a new try  without quotes in the last 3 lines of my grott.ini

**Reply by ZonPV** [reply]
Don't know where I read to use quotes. But that helped. No errors in the log any more. But the application stops after

`- Grott InfluxDB V2 initiating started` 
and publishing the Grott settings.. No futher explanation..

**Comment by rany2** [comment]
InfluxDB 2 works fine for me. Make sure that your org, bucket and token is correct. They are case-sensitive. 

If org and bucket are correct, make sure that the token was given the permissions properly or just use the all-access token type.

**Reply by ZonPV** [reply]
@rany2 Thanks for reply.
Got the application started without errors in the log......
Confirmation in the log: 
` Grott InfluxDB V2 initiating started`

And this is the part in the log about Influxdb:
```
_Influxdb:
	influx:             	 True
	influx2:             True
	database:         grottdb
	ip:                 	 192.168.1.18
	port:               	 8087
	user:               	 grott
	password:         **secret**
	organization:    xxxxxxxx
	bucket:             Grott
	token:               **secret**
```
But that's it. Nothing else happens... No input from source/inverter or uploads to any databases.. No errors in the log of container Influxdb v2.6.1

Back to Version 1 for this moment. :( 
Running version 2.7.6. One last try with Grottserver 2.7.8 to go.

**Comment by ZonPV** [comment, marked as answer]
Yeah!!! During a short break I had a clear moment!  
Went to the inverter and noticed a red light on the adapter. Happened before.... Most of the time a restart of the container with Grott will solve the problem.
Now it didn't. So I shut down the inverter from the grid.
And there it goes. First with the grott.ini for Influxdb V1. And later on, after renaming the grott.ini and a restart now for Influxdb V2.
So problem solved.
Don't know the exact reason what was wrong... The quotation marks as used in my grott.ini in the first post was wrong anyway..
Thanks for the replies.
